graphql/schemabuilder/helpers.go

```diff
@@ -372,9 +343,13 @@ func (sb *schemaBuilder) buildPaginatedField(typ reflect.Type, f interface{}) (*
 
 	var hasRet, hasError bool
 
+	var nodeType reflect.Type
 	if len(out) > 0 && out[0] != errType {
+		nodeType = out[0].Elem()
 		hasRet = true
 		out = out[1:]
+	} else {
+		return nil, fmt.Errorf("paginated function must have a return argument")
```

Comment:
i think we should also add some register-time checking that the return type is a slice of something